### PR TITLE
AMIGAOS: clean up compiler flags

### PR DESCRIPTION
**Pull request by raziel-**
tested locally and working

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -2841,34 +2841,35 @@ case $_host_os in
 		append_var LIBS "-lcitro3d"
 		;;
 	amigaos*)
-		# Leave out resources by default to save binary size
-		_builtin_resources=no
 		# AmigaOS (PPC) target
 		_port_mk="backends/platform/sdl/amigaos/amigaos.mk"
 		add_line_to_config_mk 'AMIGAOS = 1'
-		append_var CXXFLAGS "-mlongcall"
```
Comment (bluegr) [inline configure:2849]: This flag has been completely removed - it is used to produce smaller executables. Is it not needed anymore?

**Comment by raziel-** [comment]
It's still used in the cross-compiler part in lines 3958ff

		ppc-amigaos)
			# PPC Linker requires this to fix relocation errors
			append_var CXXFLAGS "-mlongcall"

			_port_mk="backends/platform/sdl/amigaos/amigaos.mk"
			;;
		ppc-morphos)
			# PPC Linker requires this to fix relocation errors
			append_var CXXFLAGS "-mlongcall"

			_port_mk="backends/platform/sdl/morphos/morphos.mk"
			;;

This was used as a test case to maybe catch random bugs/crashes in native builds.
Since it doesn't produced anything (i don' t know about smaller executables, but see below) i'm taking it out of the default builds process.

According to https://gcc.gnu.org/onlinedocs/gcc/PowerPC-Function-Attributes.html

> 6.33.24 PowerPC Function Attributes
These function attributes are supported by the PowerPC back end:

> longcall
> shortcall
> The longcall attribute indicates that the function might be far away from the call site and require a different (more expensive) calling sequence. The shortcall attribute indicates that the function is always close enough for the shorter calling sequence to be used. These attributes override both the -mlongcall switch and the #pragma longcall setting.

> See [IBM RS/6000 and PowerPC Options](https://gcc.gnu.org/onlinedocs/gcc/RS_002f6000-and-PowerPC-Options.html), for more information on whether long calls are necessary.

it adds a "more expensive" call sequence, which **i** translate to "slower" :-)

File: configure [hunk 2]
```diff
@@ -7233,7 +7233,6 @@ case $_host_os in
 		# and dead code elimination to reduce binary size
 		if test "$_release_build" = yes; then
 			append_var CXXFLAGS "-flto"
-			append_var CXXFLAGS "-ffunction-sections -fdata-sections"
```
Comment (bluegr) [inline configure:7236]: These flags also reduce the binary size, by removing unused code and data sections. Are they not needed anymore?

**Comment by raziel-** [comment]
According to https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html

both these options
> Together with a linker garbage collection (linker --gc-sections option) these options may lead to smaller **statically-linked** executables (after stripping).

since the native target is optimized for plugins and dynamic build, binary size isn't a problem anymore, since everything is laid out where possible

File: configure [hunk 3]
```diff
@@ -4821,7 +4822,6 @@ PLUGIN_LDFLAGS		+= -Wl,-T$(srcdir)/backends/plugins/3ds/plugin.ld -march=armv6k
 		append_var CXXFLAGS "-fPIC"
 		append_var LIBS "-use-dynld"
 		append_var LIBS "-lauto"
-		append_var _strip "-x"
```
Comment (bluegr) [inline configure:4824]: This flag is also removed

**Comment by raziel-** [comment]
yep, that was also a try to catch random crashes, but it doesn't help in eithercway, so back to default

im sorry for all the time you have to put into this pr...i should've commented on the changes in the main post :-/

**Comment by bluegr** [comment]
Great! Thanks for the feedback. Merging

**Comment by raziel-** [comment]
thank you

**Pull request merged**